File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> levelOrder = new ArrayList<>();
        if(root == null) return levelOrder;
        Queue<Node> q = new LinkedList<>();
        q.add(root);
        while(!q.isEmpty()) {
            int n = q.size();
            List<Integer> level = new ArrayList<>();
            for(int i = 0; i < n; ++i) {
                Node current = q.poll();
                level.add(current.val);
                for(var child : current.children)
                    q.add(child);
            }
            levelOrder.add(level);
        }
        return levelOrder;
    }
}

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""
from queue import Queue
class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        levelOrder = []
        if not root: return levelOrder
        q = Queue()
        q.put(root)
        while q.qsize():
            n = q.qsize()
            level = []
            for i in range (n):
                current = q.get()
                level.append(current.val)
                for child in current.children:
                    q.put(child)
            levelOrder.append(level)
        return levelOrder